modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/SegmentedProgressIndicator.kt

```diff
@@ -0,0 +1,78 @@
+package au.com.shiftyjelly.pocketcasts.endofyear
+
+import androidx.annotation.FloatRange
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.focusable
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.progressSemantics
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.unit.dp
+
+private val StrokeWidth = 2.dp
+private val GapWidth = 8.dp
+private val SegmentHeight = StrokeWidth
+private const val IndicatorBackgroundOpacity = 0.24f
+
+@Composable
+fun SegmentedProgressIndicator(
+    @FloatRange(from = 0.0, to = 1.0) progress: Float,
+    modifier: Modifier = Modifier,
+    color: Color = Color.White,
+    backgroundColor: Color = color.copy(alpha = IndicatorBackgroundOpacity),
+    numberOfSegments: Int,
+) {
```

Comment:
Most of the logic is taken from `Compose` internal implementation of [LinearProgressIndicator](https://developer.android.com/reference/kotlin/androidx/compose/material/package-summary#LinearProgressIndicator(kotlin.Float,androidx.compose.ui.Modifier,androidx.compose.ui.graphics.Color,androidx.compose.ui.graphics.Color)).

It doesn't support RTL animation right now.